### PR TITLE
Update prow to v20180125-b72c696dc

### PR DESCRIPTION
**Pull request by fejta**
```
ls cmd | grep -v -E 'bran|images|phony|mkpj|crier|config' | xargs ./bump.sh
ls cmd | grep -v -E 'bran|images|phony|mkpj|crier|config' | xargs -I '{}' echo '{}-image' | xargs make
```

/assign @BenTheElder @cjwagner

### CODE REVIEW

File: prow/Makefile [hunk 1]
```diff
@@ -20,25 +20,25 @@ ALPINE_VERSION           ?= 0.1
 # GIT_VERSION is the version of the alpine+git image
 GIT_VERSION              ?= 0.1
 # HOOK_VERSION is the version of the hook image
-HOOK_VERSION             ?= 0.198
+HOOK_VERSION             ?= v20180125-b72c696dc
```
Comment (stevekuznetsov) [inline prow/Makefile]: Why not generate this version at run-time? Without using the date, too? Then, no poor shmuck will ever need to rebase to update this file and the deployments can continue to pinpoint versions that are simple to understand and discover in the codebase.

Comment (fejta) [reply prow/Makefile]: Agreed, can I do that in a follow up PR? I've been meaning to make that change for awhile.

Why without the date? IMO the date is useful info

Comment (stevekuznetsov) [reply prow/Makefile]: Why is code from SHA `x` versioned differently if it's built today versus tomorrow? I guess whatever tool you're using to build container images can be different between today and tomorrow?

Comment (fejta) [reply prow/Makefile]: a) It is not immediately obvious that `b72c696dc` is ancient without a lot of investigation, whereas it would be pretty obvious looking at `v20150922-b72c696dc` that this image could use a refresh

b) Dockerfile scripts are not reproducible.

* There is no guarantee that we `COPY plank /plank` the same thing, since you and I might compile plank with different versions of golang.
* Every time we run `make hook-image` we also push a new version of the git and alpine images.
  - There is no guarantee that `FROM alpine:3.6` hasn't changed,
  - nor that `RUN apk add` does the same thing
  - nor that `update-ca-certificates` results in the same thing

Basically when this ran is important info

Comment (BenTheElder) [reply prow/Makefile]: :+1: to date-sha for non-reproducible builds. We use this for the test images and it works well, both of these are very useful to know.

Comment (BenTheElder) [reply prow/Makefile]: And like @fejta said above, these are *not* reproducible. Docker tags are mutable references to images etc...

Comment (BenTheElder) [reply prow/Makefile]: Side note: the go builds for these images use `go build` which isn't either, it leaks a bunch of strings from the host into the build, you can see host paths in a `panic()` for example
